### PR TITLE
Bump CLI to v3.2.1

### PR DESCRIPTION
**Pull request by wadhwa1**
# Description

Bumping up pinned cli version for cp 7.4 from 2.28.1 to 3.2.1
Reference thread - https://confluent.slack.com/archives/C010Y0EP5MZ/p1675164016491029
Another related PR - https://github.com/confluentinc/packaging/pull/1049

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
https://jenkins.confluent.io/job/cp-ansible-on-demand/1130/ ✅ 

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Review by utkarsh5474 — APPROVED** [review]

**Review by nsharma-git — APPROVED** [review]

File: docs/VARIABLES.md [hunk 1]
```diff
@@ -456,7 +456,7 @@ Default:  "/usr/local/bin/confluent"
 
 Confluent CLI version to download (e.g. "1.9.0"). Support matrix https://docs.confluent.io/platform/current/installation/versions-interoperability.html#confluent-cli
 
-Default:  2.28.1
+Default:  3.2.0
```
Comment (brianstrauch) [inline docs/VARIABLES.md:459]:
```suggestion
Default:  3.2.1
```

File: roles/variables/defaults/main.yml [hunk 2]
```diff
@@ -195,7 +195,7 @@ confluent_cli_base_path: /opt/confluent-cli
 confluent_cli_path: "/usr/local/bin/confluent"
 
 ### Confluent CLI version to download (e.g. "1.9.0"). Support matrix https://docs.confluent.io/platform/current/installation/versions-interoperability.html#confluent-cli
-confluent_cli_version: 2.28.1
+confluent_cli_version: 3.2.0
```
Comment (brianstrauch) [inline roles/variables/defaults/main.yml:198]:
```suggestion
confluent_cli_version: 3.2.1
```

**Review by brianstrauch — CHANGES_REQUESTED** [review]
Please bump to v3.2.1!

**Review by brianstrauch — APPROVED** [review]

**Comment by wadhwa1** [comment]
> Please bump to v3.2.1!

Thanks Brian for the review! :) 
currently, tests are failing for -
```
11:00:58  TASK [confluent.platform.common : Expand remote Confluent CLI archive] *********
11:01:03  fatal: [zookeeper32]: FAILED! => {"changed": false, "msg": "Failure downloading https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/3.2.1/confluent_v3.2.1_linux_amd64.tar.gz, HTTP Error 404: Not Found"}
```
I believe, once this PR https://github.com/confluentinc/packaging/pull/1049 is merged, we'll have this archive available in s3 bucket ??

**Comment by brianstrauch** [comment]
@wadhwa1 Please remove the leading "v" in the version number! `confluent_v3.2.1_linux_amd64.tar.gz` should become `confluent_3.2.1_linux_amd64.tar.gz`.

**Comment by wadhwa1** [comment]
> @wadhwa1 Please remove the leading "v" in the version number! `confluent_v3.2.1_linux_amd64.tar.gz` should become `confluent_3.2.1_linux_amd64.tar.gz`.

ahh okay, that's little strange.. because all the while we have been prepending 'v' in front of cli version (in cp-ansible code)
```
### A path reference to a local archive file or URL. By default this is the URL from Confluent CLI repository.
confluent_cli_archive_file_source: "{{confluent_cli_repository_baseurl}}/confluent-cli/archives/{{confluent_cli_version}}/{{confluent_cli_binary}}_{{(confluent_cli_version == 'latest') | ternary('', 'v')}}{{confluent_cli_version}}_{{ansible_system|lower}}_{{confluent_cli_goarch[ansible_architecture]}}.tar.gz”
```
Has this changed recently and is this the new convention ?

**Comment by brianstrauch** [comment]
@wadhwa1 Yes, all versions of the CLI 3.x and above no longer have the "v" in the archive name. It's an industry standard and what customers expect!

**Comment by wadhwa1** [comment]
> @wadhwa1 Yes, all versions of the CLI 3.x and above no longer have the "v" in the archive name. It's an industry standard and what customers expect!

perfect, thanks for clarifying @brianstrauch ! 👍

**Pull request merged**